pride-core/src/main/java/com/prezi/pride/vcs/RepoCache.java

```diff
@@ -42,7 +45,17 @@ public void checkoutThroughCache(VcsSupport vcsSupport, final String repositoryU
 			vcsSupport.checkout(repositoryUrl, moduleInCache, null, false, true);
 		} else {
 			log.info("Updating cached repository in " + moduleInCacheName);
-			vcsSupport.update(moduleInCache, null, false, true);
+			try {
+				vcsSupport.update(moduleInCache, null, false, true);
+			} catch (PrideException e) {
+				/*
+				 * Probably happended because mercurial repos (origin vs. repo in cache)
+				 * are unrelated. Remedy by reinitializing the cache.
+				 */
+				log.info("Cleaning up stale repo ({}) from the cache: {}", repositoryUrl, moduleInCacheName);
+				FileUtils.deleteDirectory(moduleInCache);
+				vcsSupport.checkout(repositoryUrl, moduleInCache, null, false, true);
+			}
```

Comment:
The exact condition is tested in `HgCacheRepoTest.groovy`. Basically what if
repo in cache and the one that you are adding become unrelated? Then the cache
should be removed first.

I tried googling if there is way to detect upfront if repos are unrelated. No
luck. As far as I can tell, you just need try and see if it'll fail or not.

I'll think about this one a little bit. Maybe there is a better way.